src/Switch-FTDI_FTD2xx/main.py

```diff
@@ -48,94 +49,113 @@ class Device(EmptyDevice):
         <p>The <em>Start Command </em>and <em>End command </em>are sent at the configure and unconfgure steps, respectively.</p>
     """
 
-    def __init__(self) -> None:
-        EmptyDevice.__init__(self)
 
-        self.shortname = "FTD2xx"
-        self.variables = ["Response"]
-        self.units = []
+def __init__(self) -> None:
+    EmptyDevice.__init__(self)
+
+    self.shortname = "FTD2xx"
+    self.variables = ["Response"]
+    self.units = []
+
+    self.port_manager = False
+
+    # Read and write timeout in ms
+    self.timeout_ms = 5000
+
+
+def find_ports(self):
+    return [dev.decode() for dev in ftd.listDevices()]
+
+
+def set_GUIparameter(self):
+    return {
+        "Encoding": ["HEX", "ASCII"],
+        "SweepMode": "Command",
+        "Start command": "",
+        "End command": "",
+        "Readout length": 10,
```

Comment:
What about using a default readout length of 0 so that people need to switch it on when really needed.